templates/prometheus.go

```diff
@@ -45,5 +53,26 @@ var PrometheusTemplate = promv1.Prometheus{
 			}},
 		},
 		Resources: utils.GetResourceRequirements("prometheus"),
+		Web:       PrometheusWebSpecTemplate.DeepCopy(),
+	},
+}
+
+var locSecretRef = corev1.LocalObjectReference{
+	Name: PrometheusServiceSecretName,
+}
```

Comment:
Shared instances are always tricky, you change in one location, and all location changes. 
Let's remove this and create a separate `LocalObjectReferance` object at each location.

Reply:
Removed